playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/PaymentActivity.java

```diff
@@ -122,13 +122,7 @@ private void onSkusQueried(BillingResult result, List<SkuDetails> skus) {
     }
 
     private void launchPriceChangeConfirmationFlow(SkuDetails sku) {
-        mWrapper.launchPriceChangeConfirmationFlow(this, sku, result -> {
-            if (result.getResponseCode() == BillingClient.BillingResponseCode.OK) {
-                setResultAndFinish(PaymentResult.priceChangeSuccess());
-            } else {
-                fail("Price change confirmation flow ended with result: " + result);
-            }
-        });
+        fail("Price change confirmation flow is not supported");
```

Comment:
I think we can fail earlier. At the moment, the `PaymentActivity` is launched, it connects to Play Billing (which is async) and queries the SKUs (which is async) then hits this, but we know in `PaymentActivity#onCreate` if the request is for a price change confirmation flow.

I think you could instead add `if (mMethodData.isPricechangeConfirmation) { fail(...) }` as soon as we create the MethodData.